### ISSUE TITLE
jest throws `Validation Warning: Unknown option "coverageDirectory"` in new, clean (Angular) repo

### ISSUE THREAD

**Issue by alexciesielski** [nrwl/nx#15752]
### Current Behavior

Running `nx test app` throws validation warnings.

```
> nx run app:test

● Validation Warning:

  Unknown option "coverageDirectory" with value "../../coverage/packages/app" was found.
  This is probably a typing mistake. Fixing it will remove this message.

  Configuration Documentation:
  https://jestjs.io/docs/configuration
```

### Expected Behavior

Don't show validation warnings on supported config properties.

### GitHub Repo

https://github.com/alexciesielski/nx-test-warning-showcase

### Steps to Reproduce

1. `npx create-nx-workspace@latest` => `package-based monorepo` => `app` => `yes`
2. Create `tsconfig.base.json` (without it generating an angular app throws an error, guess this is also a bug?)
3. `npm i -D @nrwl/angular`
4. `npx nx g @nrwl/angular:application --name=app`
5. `nx test app` => warning in console

### Nx Report

```shell
nx report

 >  NX   Report complete - copy this into the issue template

   Node : 18.15.0
   OS   : darwin arm64
   npm  : 9.5.0

   nx                      : 15.8.7
   @nrwl/js                : 15.8.7
   @nrwl/jest              : 15.8.7
   @nrwl/linter            : 15.8.7
   @nrwl/workspace         : 15.8.7
   @nrwl/angular           : 15.8.7
   @nrwl/cli               : 15.8.7
   @nrwl/cypress           : 15.8.7
   @nrwl/devkit            : 15.8.7
   @nrwl/eslint-plugin-nx  : 15.8.7
   @nrwl/storybook         : 15.8.6
   @nrwl/tao               : 15.8.7
   @nrwl/webpack           : 15.8.7
   typescript              : 4.9.5
   ---------------------------------------
   The following packages should match the installed version of nx
     - @nrwl/storybook@15.8.6

   To fix this, run `nx migrate nx@15.8.7`
```

### Failure Logs

```shell
> nx run app:test

● Validation Warning:

  Unknown option "coverageDirectory" with value "../../coverage/packages/app" was found.
  This is probably a typing mistake. Fixing it will remove this message.

  Configuration Documentation:
  https://jestjs.io/docs/configuration

 PASS   app  packages/app/src/app/app.component.spec.ts
  AppComponent
    ✓ should create the app (62 ms)
    ✓ should have as title 'app' (9 ms)
    ✓ should render title (9 ms)

Test Suites: 1 passed, 1 total
Tests:       3 passed, 3 total
Snapshots:   0 total
Time:        1.738 s
Ran all test suites.
```

**Comment by alexciesielski** [comment]
I've now also attached a repro repo [here](https://github.com/alexciesielski/nx-test-warning-showcase)

**Comment by nemonemi** [comment]
I can confirm to have started seeing this after updating to the latest version of the NX.

**Comment by nikbabchenko** [comment]
The same for me. 
And for some reason there is no coverage report in terminal when executing command - 'nx run-many --all --target=test --coverage=true --coverageReporters='cobertura,lcov,text'' 

or just 'nx run-many --all --target=test --coverage=true'

**Comment by bojanbass** [comment]
+1

**Comment by SaulMoro** [comment]
+1

**Comment by barbados-clemens** [comment]
This is a bug in jest 29, 
If you're not using code coverage you can remove the property for now,
But you can safely ignore this warning since the property is working. 

https://github.com/facebook/jest/issues/13576

**Comment by bill-mcgrory** [comment]
If you run npx jest in the given module directory, it does not give the warnings, plus it honors the jest.config.ts file. I am having what I think is a related issue, where the reporters key is ignored (and reported as a syntax error) if I run nx test module, but if am in the module root directory, it honors the config settigns for reporters and does not complain about syntax errors.

**Comment by Fitzbert-Fitz** [comment]
we also see that no reports are generated when we see the warning. Removing the coverageDirectory fixes this for us and reports are generated again.

**Comment by ojemuyiwa** [comment]
Hi all 👋 any fix for this yet?

**Comment by rhutchison** [comment]
https://github.com/jestjs/jest/pull/14180#issuecomment-1620447618

**Comment by LayZeeDK** [comment]
@AgentEnder Reopen this issue because of https://github.com/nrwl/nx/pull/18096

**Comment by barbados-clemens** [comment]
going to leave closed as you can still manually bump to v29.6.x on your own. we ran into some odd memory issues so reverted in attempt to track down. 
`@nx/jest` package show work with 29.x.x version.
